engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -278,4 +284,63 @@ public static synchronized void changeLoggersLevel(@Nonnull Map<String, String>
     }
     return ImmutableMap.copyOf(matchedClasses);
   }
+
+  /**
+   * Returns logger log file name
+   * @return logFileName
+   */
+  @Evolving
+  public static String getLogFileName()
+  {
+    Set<FileAppender> fileAppenders = getRootFileAppenders();
+    //skip fetching log file name if we have multiple file Appenders
+    if (fileAppenders.size() == 1) {
+      FileAppender fileAppender = fileAppenders.iterator().next();
+      if (isErrorLevelEnable(fileAppender)) {
+        return fileAppender.getFile();
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Returns offset of current log file.
+   * @return logFileOffset
+   */
+  @Evolving
+  public static long getLogFileOffset()
+  {
+    Set<FileAppender> fileAppenders = getRootFileAppenders();
+    //skip fetching log file offset if we have multiple file Appenders
+    if (fileAppenders.size() == 1) {
+      FileAppender fileAppender = fileAppenders.iterator().next();
+      if (isErrorLevelEnable(fileAppender)) {
+        File logFile = new File(fileAppender.getFile());
+        return logFile.length();
```

Comment:
Only option I see to always enable ability to flush is to override shouldFlush and return true always. But for this, we will need to extend FileAppender and introduce a custom appender which isn't very good solution.
As the flag "immediateFlush" can be changed from config file, as of now there is no 100% gaurentee of keeping it enable all times (by default it's enable)

We can skip feature if flag is false, but again as we are giving approx offset before exception, I think we need not disable feature for "immediateFlush = false". Let me know what you think.